### DISCUSSION TITLE
Voicemail to Case failure.  Failed to authenticate with Salesforce.

### DISCUSSION THREAD

**Discussion by wlee598** [amazon-connect/amazon-connect-salesforce-scv#99] [Q&A] [answered]
Hello,
When I viewed the "Code source" in VMXPackager, I noticed that AWS is failing to authenticate with Salesforce (image below).
![Failed to authenticate with Salesforce](https://user-images.githubusercontent.com/103658462/164535800-760fee6b-e6b9-4013-b168-729c225e2c16.png)
Could this be an issue with my .crt file?  When I created the self-signed SSL in OpenSSL (in windows), base64 was not a recognized command.  I had to use certutil to complete this process but may have messed it up.  If this is the problem, how can I create the self-signed SSL in Windows?

This is my first of 7 errors on this screen, but it seems it could be a root cause of the errors down the line.  
Thank you.

**Comment by dougjaso** [comment, marked as answer]
Make sure that your [Salesforce Validator Lambda function works](https://github.com/amazon-connect/amazon-connect-salesforce-scv/blob/master/Common/AWSSCV-SalesforceConfig/Docs/installation.md#post-install-validation) before attempting to use VMX. If it does, then you should not encounter this error. 

The instructions provided were based on Salesforce's instructions on how to [Create a Private Key and Self-Signed Digital Certificate ](https://developer.salesforce.com/docs/atlas.en-us.228.0.sfdx_dev.meta/sfdx_dev/sfdx_dev_auth_key_and_cert.htm). 

You can get the [Windows binaries for OpenSSL](https://wiki.openssl.org/index.php/Binaries) from the OpenSSLWiki. I have not used the Windows version, so I cannot verify that they function exactly the same.

**Reply by wlee598** [reply]
Thank you Doug.  I tested the AWSSCV_Salesforce_Validator and received a confusing response.  The execution result was "succeeded", but the logs show { "Status": "Failure", "Reason": "Incorrect Padding" }
![Validator Succeeded but Failure](https://user-images.githubusercontent.com/103658462/164723119-4546cfe7-53ab-4eff-9d2b-c08b97488d12.png)
I've done a number of searches on the meaning of this error and "base64" is repeatedly mentioned.  This issue has thoroughly stumped me.  If it is related to the Private Key and Self-Signed Digital Certificate, might I have better luck creating those on a Linux machine?
Thank you again.

**Reply by dougjaso** [reply]
The function code executed correctly (the code caught and reported the error as designed), the result of the code was a failure, in this case from Salesforce. Think of it like this: your function worked, you just didn't get the response you were hoping for. 

Incorrect padding tends to point to an issue with the base64 encoded key. You can validate the contents of your encoded key by using a tool, such as https://www.base64decode.org/ to see if the content matches.

**Reply by wlee598** [reply]
Thank you for the help!  I got the validator to succeed.  
![image](https://user-images.githubusercontent.com/103658462/164760870-0486260a-a4c1-4d85-8182-025267b34c80.png)
After I got the success message, I did another test vm to case.  After the case was not created, I checked my Cloudwatch logs and there are two errors in my most recent attempt.  The first says "list indices must be in integers or slices, not str".  The second error just says "Record 1 Result: Failed to create Salesforce case".  I've combed through the installation guide but can't find what might be the issue.
![Cloudwatch Error SF Case](https://user-images.githubusercontent.com/103658462/164761678-dfb20fac-8225-4573-8b98-2deb1e49ab70.png)
Thank you

**Reply by dougjaso** [reply]
Typically, this results from one of the custom field API names being incorrect. Make sure you have [added all the fields](https://github.com/amazon-connect/amazon-connect-salesforce-scv/blob/master/Solutions/VMX2-VoicemailExpress/Docs/vmx_prerequistes.md#setup-steps-for-all-salesforce-models) correctly. If you have, send me an email at dougjaso@amazon.com and I'll work with you to find a time to troubleshoot live.

**Reply by dougjaso** [reply]
Addressed via troubleshooting session. Voicemail link field missing from case object. Had been added to custom object instead. Corrected in org, solution operational.

**Reply by wlee598** [reply]
Thanks for all your help!
